apps/nextjs-website/src/components/organisms/GuideInPageMenu/components/Heading.tsx

```diff
@@ -0,0 +1,43 @@
+import { HeadingProps } from 'gitbook-docs/markdoc/schema/heading';
+import React, { ReactNode } from 'react';
+import MUILink from '@mui/material/Link';
+import { removeEmojis } from '@/markdoc/helpers';
+import { Typography, useTheme } from '@mui/material';
+import { useHash } from '@/components/organisms/HashProvider/HashProvider';
+
+const Heading = ({ level, id, children }: HeadingProps<ReactNode>) => {
+  const isString = typeof children === 'string';
+  const { palette } = useTheme();
+  const { hash } = useHash();
+
+  const isCurrentHash = hash === `#${id}`;
+
+  if (level < 2 || level > 4) {
+    return;
+  }
+
+  return (
+    <MUILink
+      href={`#${id}`}
+      title={isString ? removeEmojis(children) : ''}
```

Comment:
I think the `removeEmojis` is useless. They are already removed during the parsing phase